src/Interpreters/convertFieldToType.h

```diff
@@ -20,4 +20,8 @@ Field convertFieldToType(const Field & from_value, const IDataType & to_type, co
 /// Does the same, but throws ARGUMENT_OUT_OF_BOUND if value does not fall into the range.
 Field convertFieldToTypeOrThrow(const Field & from_value, const IDataType & to_type, const IDataType * from_type_hint = nullptr);
 
+/// Applies stricter rules than convertFieldToType, doesn't allow loss of precision converting to Decimal
+/// Returns true if the conversion was successful and the result is equal to the original value
+bool convertFieldToTypeStrict(const Field & from_value, const IDataType & to_type, Field & result_value);
```

Comment:
Better to return `pair<bool, Field>` or even to introduce a separate structure. It can be used like `auto [success, field] = convertFieldToTypeStrict(...)`.

Reply:
Changed to return `std::optional<Field>`